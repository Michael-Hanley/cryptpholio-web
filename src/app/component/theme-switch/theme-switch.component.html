<mat-slide-toggle
    class="theme-toggle"
    (click)="changeTheme()"
    [checked]="darkMode">
</mat-slide-toggle>