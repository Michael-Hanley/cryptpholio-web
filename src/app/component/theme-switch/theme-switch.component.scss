.theme-toggle {
    margin: 10px 20px;
  }