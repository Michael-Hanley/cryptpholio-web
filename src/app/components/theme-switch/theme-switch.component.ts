import { Component, OnInit } from '@angular/core';
import { ThemeService } from '../../services/theme.service';

@Component({
  selector: 'app-theme-switch',
  templateUrl: './theme-switch.component.html',
  styleUrls: ['./theme-switch.component.scss']
})
export class ThemeSwitchComponent implements OnInit {
  currentTheme = 'light-theme';
  darkMode = false;

  constructor(private themeService: ThemeService) { }

  changeTheme() {
    this.currentTheme = this.currentTheme !== 'dark-theme' ? 'dark-theme' : 'light-theme';
    this.themeService.updateThemeObject(this.currentTheme);
  }

  ngOnInit() {
    this.currentTheme = this.themeService.currentTheme;
    this.darkMode = this.currentTheme === 'dark-theme' ? true : false;
  }

}
