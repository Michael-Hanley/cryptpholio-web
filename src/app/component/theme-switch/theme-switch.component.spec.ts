import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { ThemeSwitchComponent } from './theme-switch.component';

import { MatSlideToggleModule } from '@angular/material/slide-toggle';
import 'hammerjs';

describe('ThemeSwitchComponent', () => {
  let component: ThemeSwitchComponent;
  let fixture: ComponentFixture<ThemeSwitchComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ ThemeSwitchComponent ],
      imports: [MatSlideToggleModule]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ThemeSwitchComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
