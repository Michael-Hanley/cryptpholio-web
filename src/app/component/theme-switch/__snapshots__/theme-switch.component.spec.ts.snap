// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitchComponent should match snapshot 1`] = `
<app-theme-switch
  currentTheme={[Function String]}
  darkMode="false"
  themeService={[Function ThemeService]}
>
  <mat-slide-toggle
    class="theme-toggle mat-slide-toggle mat-accent"
    id="mat-slide-toggle-2"
    tabindex="-1"
  >
    <label
      class="mat-slide-toggle-label"
      for="mat-slide-toggle-2-input"
    >
      <div
        class="mat-slide-toggle-bar"
      >
        <input
          aria-checked="false"
          class="mat-slide-toggle-input cdk-visually-hidden"
          id="mat-slide-toggle-2-input"
          role="switch"
          tabindex="0"
          type="checkbox"
        />
        <div
          class="mat-slide-toggle-thumb-container"
        >
          <div
            class="mat-slide-toggle-thumb"
          />
          <div
            class="mat-slide-toggle-ripple mat-ripple"
            mat-ripple=""
          >
            <div
              class="mat-ripple-element mat-slide-toggle-persistent-ripple"
            />
          </div>
        </div>
      </div>
      <span
        class="mat-slide-toggle-content"
      >
        <span
          style="display:none"
        >
          Â 
        </span>
         Dark Theme
  
      </span>
    </label>
  </mat-slide-toggle>
</app-theme-switch>
`;
